### PR TITLE
switch app-policy to go modules from glide

### PR DESCRIPTION
**Pull request by rafaelvanoni**

### CODE REVIEW

**Comment by spikecurtis** [comment]
This does *not* appear to switch to go modules; there is no `go.mod`.  Are you missing some commits in this PR?

File: Makefile [hunk 1]
```diff
@@ -157,80 +191,63 @@ build-all: $(addprefix bin/dikastes-,$(VALIDARCHES))
...
-	mkdir -p $(HOME)/.glide
-
-	# To build without Docker just run "glide install -strip-vendor"
-	docker run --rm -i \
-      -v $(CURDIR):/go/src/$(PACKAGE_NAME):rw \
-      -v $(HOME)/.glide:/home/user/.glide:rw \
-      -e LOCAL_USER_ID=$(LOCAL_USER_ID) \
-      -w /go/src/$(PACKAGE_NAME) \
-      $(CALICO_BUILD) glide install -strip-vendor
+.PHONY: vendor
```
Comment (spikecurtis) [inline Makefile]: does vendor need to be `.PHONY`? It makes basically every target re-download.

File: Makefile [hunk 2]
```diff
@@ -157,80 +191,63 @@ build-all: $(addprefix bin/dikastes-,$(VALIDARCHES))
 
 .PHONY: build
 ## Build the binary for the current architecture and platform
-build: bin/dikastes-$(ARCH) bin/healthz-$(ARCH)
+build: vendor bin/dikastes-$(ARCH) bin/healthz-$(ARCH)
```
Comment (spikecurtis) [inline Makefile]: why the explicit dependence here? I believe that each binary target already has dependencies on vendor

File: Makefile [hunk 3]
```diff
@@ -157,80 +191,63 @@ build-all: $(addprefix bin/dikastes-,$(VALIDARCHES))
...
-          glide up --strip-vendor 2>&1 | tee $$OUTPUT; \
-          if ! grep "\[WARN\]" $$OUTPUT; then true; else false; fi; \
-        fi'
+	$(DOCKER_RUN) -i $(CALICO_BUILD) sh -c '\
+	if [ $(LIBCALICO_VERSION) != $(LIBCALICO_OLDVER) ]; then \
+		echo "Updating libcalico version $(LIBCALICO_OLDVER) to $(LIBCALICO_VERSION) from $(LIBCALICO_REPO)"; \
+		go mod edit -droprequire github.com/projectcalico/libcalico-go; \
+		go get $(LIBCALICO_REPO)@$(LIBCALICO_VERSION); \
+		if [ $(LIBCALICO_REPO) != "github.com/projectcalico/libcalico-go" ]; then \
+			go mod edit -replace github.com/projectcalico/typha=$(LIBCALICO_REPO)@$(LIBCALICO_VERSION); \
```
Comment (spikecurtis) [inline Makefile]:
```suggestion
			go mod edit -replace github.com/projectcalico/libcalico-go=$(LIBCALICO_REPO)@$(LIBCALICO_VERSION); \
```

File: Makefile [hunk 4]
```diff
@@ -374,13 +391,13 @@ sub-tag-images-%:
 # Static checks
 ###############################################################################
 ## Perform static checks on the code.
+
+# TODO: re-enable these linters !
```
Comment (spikecurtis) [inline Makefile]: I think that TODO line is a stray paste

File: Makefile [hunk 5]
```diff
@@ -420,7 +434,7 @@ ci: clean build-all static-checks ut
 ###############################################################################
 .PHONY: cd
 ## Deploys images to registry
-cd: image-all
+cd: vendor image-all
```
Comment (spikecurtis) [inline Makefile]: why the explicit vendor dependence?

File: Makefile (hunk 1 above)
Comment (rafaelvanoni) [reply Makefile]: Removed

File: Makefile (hunk 2 above)
Comment (rafaelvanoni) [reply Makefile]: Fixed

File: Makefile (hunk 3 above)
Comment (rafaelvanoni) [reply Makefile]: Fixed

File: Makefile (hunk 5 above)
Comment (rafaelvanoni) [reply Makefile]: Fixed

File: Makefile (hunk 4 above)
Comment (rafaelvanoni) [reply Makefile]: Fixed

**Pull request merged**